src/data/var/acceptData.js

```diff
@@ -7,7 +7,9 @@ export default function( owner ) {
 	//  - Node
 	//    - Node.ELEMENT_NODE
 	//    - Node.DOCUMENT_NODE
+	//    - Node.SHADOWROOT_NODE
 	//  - Object
 	//    - Any
-	return owner.nodeType === 1 || owner.nodeType === 9 || !( +owner.nodeType );
+	return owner.nodeType === 1 || owner.nodeType === 9 ||
+		( owner.nodeType === 11 && owner.host ) || !( +owner.nodeType );
```

Comment:
@jquery/core do you remember the reasons we don't allow data on document fragments? Or in general, what are the reasons to maintain any restrictions? In the past there were issues with host objects, browser plugins, etc., but plugins went away and many of these problems got smoothed out.